File: RemoteMonitor Desktop/src/main/java/components/PhysicalDisk.java
package components;

import com.profesorfalken.jsensors.model.sensors.Load;
import oshi.hardware.HWDiskStore;
import oshi.hardware.HWPartition;
import com.profesorfalken.jsensors.model.components.Disk;
import java.io.File;
import java.util.List;

public class PhysicalDisk extends Component {
    private long capacity;
    private long availableCapacity;
    private transient List<HWPartition> partitions;
    private transient HWDiskStore diskStore;
    private transient Disk jdisk;
    public PhysicalDisk(HWDiskStore hwDiskStore, Disk Jdisk)
    {
        diskStore = hwDiskStore;
        capacity = diskStore.getSize();
        componentName = diskStore.getModel();
        componentType = "Disk";
        partitions = diskStore.getPartitions();
        jdisk = Jdisk;
    }

    public void update()
    {
        partitions = diskStore.getPartitions();
        setAvailableCapacity();
        temperature = checkTemp(jdisk.sensors.loads);
        lastUpdated = System.currentTimeMillis();
    }
    public long getAvailableCapacity() {
        return availableCapacity;
    }

    public long getCapacity() {
        return capacity;
    }

    private void setAvailableCapacity()
    {
        availableCapacity = 0;
        for(HWPartition p : partitions)
        {
            File diskPartition = new File(p.getMountPoint());
            availableCapacity += diskPartition.getFreeSpace();
        }
    }

    //checks if disk temperature can be accessed and returns its value
    private double checkTemp(List<Load> loads)
    {
        for (Load l : loads)
        {
            if (l.name.contains("Temperature")){
                return l.value;
            }
        }
        return 0;
    }
}

File: RemoteMonitor Mobile/RemoteMonitor/RemoteMonitor/RemoteMonitor/Components/PhysicalDisk.cs
using Newtonsoft.Json.Linq;
using System;
using System.Collections.Generic;
using System.Text;

namespace RemoteMonitor.Components
{
    public class PhysicalDisk : Component 
    {
        public long capacity;
        public long availableCapacity;

        public PhysicalDisk() 
        {
            componentName = "No Disk detected";
            componentType = "Disk";
            temperature = 0;
            capacity = 0;
            availableCapacity = 0;
        }
        public PhysicalDisk(string name, double temp, long diskCapacity, long availableDiskCapacity)
        {
            componentName = name;
            componentType = "Disk";
            temperature = temp;
            capacity = diskCapacity;
            availableCapacity = availableDiskCapacity;
            update(temp, diskCapacity, availableDiskCapacity);
        }
        public void update(double temp, long diskCapacity, long availableDiskCapacity)
        {
            temperature = temp;
            capacity= diskCapacity;
            availableCapacity= availableDiskCapacity;
            if(capacity != 0)
                usage = availableCapacity / capacity;
        }
        public PhysicalDisk(JObject json)
        {
            componentName = json["componentName"].ToString();
            componentType = "Disk";
            temperature = json["temperature"].ToObject<double>();
            capacity = json["capacity"].ToObject<long>();
            availableCapacity = json["availableCapacity"].ToObject<long>();
            usage = json["usage"].ToObject<double>();
            update(json);
        }
        public void update(JObject json)
        {
            temperature = json["temperature"].ToObject<double>();
            capacity = json["capacity"].ToObject<long>();
            availableCapacity = json["availableCapacity"].ToObject<long>();
            usage = json["usage"].ToObject<double>();
        }
        public long getAvailableCapacity()
        {
            return availableCapacity;
        }

        public long getCapacity()
        {
            return capacity;
        }
    }
}
